README.md

```diff
@@ -42,6 +42,38 @@ The `flag` is optional (if provided, it must be surrounded in square brackets).
 
 Exclude selected commit types (comma separated).
 
+#### `generate`
+
+Generate or update a change log file.
+
+#### `path`
+
+If `generate` set, specifies the path of the change log file to update (default `./CHANGELOG.md`)
+
+#### `title`
+
+If `generate` set, specifies the markdown text to use as the file heading (default `# Changelog`)
+
+#### `section`
+
+If `generate` set, specifies the tagged markdown text to use for each release section (default `## Release {{ GITHUB_REF }}`)
+
+#### `files_to_commit`
+
+Commits any files, like the updated changelog file, to the repo on completion (comma separated). Does not commit anything by default.
+
+#### `commit_message`
+
+If `files_to_commit` contains any files to commit, this specifies the message to use (default `chore(pipeline updates): [skip ci]`)
+
+#### `author_name`
+
+If `files_to_commit` contains any files to commit, this specifies the Author Name to assign to the commit (required if files are being committed)
+
+#### `author_email`
+
+If `files_to_commit` contains any files to commit, this specifies the Author Email to assign to the commit (required if files are being committed)
+
```

Comment:
Why don't we just add one option called `file` or something and handle the rest on our own
And also modify the changelog file in a way that the user itself can modify the headers in separate commits if they want, and we just add the logs below the current headers and new ones upper the old logs

Reply:
The reason why I added some options was that every circumstance will be a bit different; I did think of using a special config file like `.changelog` that had these options and that we defaulted with some reasonable defaults for everything else. I would be happy to make changes to use such a file or remove them as you suggest; but if no configuration is possible I would like your permission to fork and release an alternative action that does include these options as I sort of need them for my personal and private projects.

For example, the defaults could be changed to be as follows, and the user can edit the file afterwards (as you suggest) if they wish to edit the section to include a version number and/or tag link:
* title = `# repository name`
* section = `## Release 2020-10-17` (the current date)
* commit_message = `chore(pipeline updates): [skip ci]`
* files_to_commit = `./CHANGELOG.md`
* author_name = `GitHub CI`
* author_email = `no idea what to default this to`

As above, there is no 'email' that can be determined in the pipeline to use for this; I guess we could take the email of the last commit but I dont know if that would be correct or useful and its not something that can be edited after the fact.

What are your thoughts?

Reply:
@SamuelJohnsonMedia I'm thinking like merging the three `generate`, `path` & `files_to_commit` into one option like `file` or something like that which would contain the `CHANGELOG.md` or `./HISTORY.md` or whatever, which might be an empty string by default, if not then internally consider `generate` to true & `files_to_commit` to be the given value.

And for the author, we can use the committer of the tag or the action initiator if possible, maybe with some posix like `- Auto-Changelog` or ` (Github CI)`, you get the idea.

And for the `commit_message` I guess we can use something like `chore: Update [the given file value here] [omit]`?

This way the necessary options are reduced and there's more automation & hopefully satisfies your needs as well

Something like [this](https://github.com/lob/generate-changelog), which is what I used as a reference

What do you think?

Reply:
Also for the header file I didn't mean it should be hand edited every time, what I meant was we would just create the file if it doesn't exist and set the title to '# Changelog' by default and the version tags like '## tag (date)' and if the users want different title like History instead, or want to have some description before the logs they can create the file before or modify it whenever they want to make the said customizations and we would follow like this:
'
[Whatever user put here]

[The generated logs here]

'

Reply:
I understand what you mean now with relation to the options; so I have modified the PR so its just a single `file` option which, if it exists, will be the file that is updated or created.

I have also modified the file content modifier to locate the last 'release' that was made and insert the new content above it; this way the user can add whatever they want above without it being removed.

I didnt change the release heading to be the tag name as the tag wont have been created at the point this action is run; it will be run afterwards so that the changelog is included in the release.

Feel free to modify if you wish.